redpanda/values.yaml

```diff
@@ -104,19 +111,33 @@ listeners:
   # Kafka API listeners
   # The kafka listener group cannot be disabled
   kafka:
-    # The first entry is used to set up the headless service.
-    # Kafka external and internal advertised listeners will be
-    # derived from the configuration. External advertised listener will be via
-    # node port unless load balancer is configured.
+    # The first entry is used to set up the headless service, which enables
+    # internal clients within the same namespace to connect directly to brokers.
+    # By default there are two listeners, one for internal and another for external clients.
+    # A listener with the name `internal` must exist, and will always be advertised
+    # using the internal address.
+    # Other listeners will have the advertised address according to whether it is
+    # externally facing or not.
     endpoints:
     # TODO POD_IP doesn't mean anything to the user
     # maybe make address optional, and if not set then default is $(POD_IP)
     - address: $(POD_IP)
-      name: default
+      name: internal
+      port: 9093
+      # TODO make tls section optional
+      tls:
+        enabled: false
```

Comment:
A kafka listener with the name `internal` is required, as this name is how the code identifies this listener as always being advertised internally rather than externally. This is because the external section is optional, so if global external access is enabled by not explicitly disabled in this listener then it would not be an internal listener without this additional logic.